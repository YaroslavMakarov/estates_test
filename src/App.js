import React, { useState, useEffect } from "react";
import 'bootstrap/dist/css/bootstrap.min.css';
import { Container, Row, Col } from "react-bootstrap";
import { getEstates } from "./helpers/api";
import Estates from "./components/Estates/Estates"
import './App.css';

const App = () => {
  const [estates, setEstates] = useState([]);

  useEffect(() => {
    getEstates().then(resolve => setEstates(resolve));
  }, []);

  console.log(estates);
  return (
    <Container className="d-flex justify-content-center">
      <Estates estates={estates} />
    </Container>
  );
}

export default App;
