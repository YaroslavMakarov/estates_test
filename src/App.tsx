import React, { useEffect, Dispatch } from 'react';
import { Container } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import {Switch, Route } from 'react-router-dom';
import { useDispatch } from 'react-redux';

import EstatesList from './components/EstatesList';
import EstateInfo from './components/EstatesInfo';

import { AppWrapper } from './styled/AppSteles';
import { AllEstatesAction, fetchPosts } from './redux/estates';


const App = () => {
  const estatesDispatch = useDispatch<Dispatch<AllEstatesAction>>();

  useEffect(() => {
    estatesDispatch(fetchPosts());
  }, []);

  return (
    <AppWrapper>
      <Container className="
          d-flex
          align-items-center
          flex-column
        "
      >
        <Switch>
          <Route
            path="/"
            exact
            component={EstatesList}
          />
          <Route
            path="/info/:id"
            exact
            component={EstateInfo}
          />
        </Switch>
      </Container>
    </AppWrapper>
  );
};

export default App;
