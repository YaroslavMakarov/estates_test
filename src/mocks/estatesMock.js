export const estatesMock = {
    estates: {
        estates: [
            {
                'id': 1,
                'title': 'Shevchenko estate',
                'price': 30000,
                'address': {
                    'street': 'Melnychna',
                    'number': 12
                },
                'preview_img': 'https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg',
                'seller': 'Chack Norris',
                'description': 'The best from Chuck',
                'detailed_photos': [
                    'https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg',
                    'https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg',
                    'https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg'
                ]
            },
            {
                "id": 2,
                "title": "Yanukovich estate",
                "price": 3000000,
                "address": {
                    "street": "Prisident",
                    "number": 4
                },
                "preview_img": "https://biz.liga.net/images/general/2018/11/15/thumbnail-tw-20181115111659-8474.jpg?v=1542279867",
                "seller": "Chack Norris",
                "description": "The best from Chuck",
                "detailed_photos": [
                    "https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg",
                    "https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg",
                    "https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
                    ]
             },
        ],
    },
};