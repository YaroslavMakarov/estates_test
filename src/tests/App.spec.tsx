import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import App from '../App';

import { estatesMock } from '../mocks/estatesMock';

describe("App component", () => {
    const mockStore = configureStore();
    let store;

    it("Shoud render App component", () => {
        store = mockStore(estatesMock);
        const component = shallow(
            <Provider store={store}>    
                <App />
            </Provider>
        ).dive();
        const app = component.find("AppWrapper");
        expect(app.length).toBe(1);
        console.log(component.debug());
    });
});

// describe("Estates component", () => {
//     it("render Estates wrapper", () => {
//         const component = shallow(<Estates />);
//         const wrapper = component.find("EstatesWrapper");
//         expect(wrapper.length).toBe(1);
//         console.log(wrapper.debug());
//     });
// });
