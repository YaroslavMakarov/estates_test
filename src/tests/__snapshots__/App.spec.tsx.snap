// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component shoud render Info component 1`] = `
"<Provider store={{...}}>
  <MemoryRouter>
    <Router history={{...}}>
      <App>
        <styled.div>
          <div className=\\"sc-jSgupP gQRInk\\">
            <Container className=\\" d-flex align-items-center flex-column \\" fluid={false}>
              <div className=\\" d-flex align-items-center flex-column  container\\">
                <Switch>
                  <Route path=\\"/\\" exact={true} component={[Function: EstatesLists]} location={{...}} computedMatch={{...}}>
                    <EstatesLists history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                      <styled.div>
                        <div className=\\"sc-hKgILt gYRjSP\\">
                          <Estate estate={{...}}>
                            <styled.div as={{...}} to=\\"/info/1\\">
                              <Link to=\\"/info/1\\" className=\\"sc-bdfBwQ fhkZDT\\">
                                <LinkAnchor className=\\"sc-bdfBwQ fhkZDT\\" href=\\"/info/1\\" navigate={[Function: navigate]}>
                                  <a className=\\"sc-bdfBwQ fhkZDT\\" href=\\"/info/1\\" onClick={[Function: onClick]}>
                                    <styled.img src=\\"https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg\\">
                                      <img src=\\"https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg\\" className=\\"sc-gsTCUz esHetz\\" />
                                    </styled.img>
                                    <styled.div>
                                      <div className=\\"sc-dlfnbm hMEdSQ\\">
                                        Name: Shevchenko estate
                                      </div>
                                    </styled.div>
                                    <styled.div>
                                      <div className=\\"sc-dlfnbm hMEdSQ\\">
                                        Price: 30000$
                                      </div>
                                    </styled.div>
                                    <styled.div>
                                      <div className=\\"sc-dlfnbm hMEdSQ\\">
                                        Address: Melnychna, 12
                                      </div>
                                    </styled.div>
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </styled.div>
                          </Estate>
                        </div>
                      </styled.div>
                    </EstatesLists>
                  </Route>
                </Switch>
              </div>
            </Container>
          </div>
        </styled.div>
      </App>
    </Router>
  </MemoryRouter>
</Provider>"
`;
